compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -13663,23 +13663,23 @@ OMR::Z::TreeEvaluator::primitiveArraycopyEvaluator(TR::Node* node, TR::CodeGener
    else
       {
       // We need to decide direction of array copy at runtime.
-      TR::LabelSymbol *forwardArrayCopyLabel = generateLabelSymbol(cg);
-      cursor = generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::getCmpRegOpCode(), node, byteSrcReg, byteDstReg, TR::InstOpCode::COND_BNL, forwardArrayCopyLabel, false);
-      iComment("if byteSrcPointer >= byteDstPointer then GoTo forwardArrayCopy");
-      
       if (isConstantByteLen)
          {
          generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, cFlowRegionStart);
          cFlowRegionStart->setStartInternalControlFlow();
          }
+      TR::LabelSymbol *forwardArrayCopyLabel = generateLabelSymbol(cg);
+      cursor = generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::getCmpRegOpCode(), node, byteSrcReg, byteDstReg, TR::InstOpCode::COND_BNL, forwardArrayCopyLabel, false);
+      iComment("if byteSrcPointer >= byteDstPointer then GoTo forwardArrayCopy");
+      
 
       TR::Register *checkBoundReg = srm->findOrCreateScratchRegister();
       if (byteLenReg == NULL)
          {
          byteLenReg = cg->gprClobberEvaluate(byteLenNode);
          }
-      cursor = generateRRRInstruction(cg, TR::InstOpCode::getAddThreeRegOpCode(), node, checkBoundReg, byteSrcReg, byteLenReg); 
-      iComment("lastElementToCopy=byteSrcPointer+lengthInBytes");
+      cursor = generateRXInstruction(cg, TR::InstOpCode::LA, node, checkBoundReg, generateS390MemoryReference(byteSrcReg, byteLenReg, 0, cg));
```

Comment:
@dhong44  This should fix the issue you were seeing on your task to enable minimum ALS asserts as fatal.